sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -821,10 +822,11 @@ case class Divide(
 
   private lazy val div: (Any, Any) => Any = dataType match {
     case d @ DecimalType.Fixed(precision, scale) => (l, r) => {
-      val value = d.fractional.asInstanceOf[Fractional[Any]].div(l, r)
+      val value =
+        PhysicalDecimalType(precision, scale).fractional.asInstanceOf[Fractional[Any]].div(l, r)
```

Comment:
can we move it outside of the lambda?

Reply:
actually we can just return `PhysicalDecimalType(precision, scale).fractional.asInstanceOf[Fractional[Any]].div`

Reply:
and we can remove `.asInstanceOf[Fractional[Any]]`